common/components/BalanceSidebar/BalanceSidebar.jsx

```diff
@@ -28,12 +28,20 @@ type Props = {
 export class BalanceSidebar extends React.Component {
   props: Props;
   state = {
-    showLongBalance: false
+    showLongBalance: false,
+    address: ''
   };
 
+  componentDidMount() {
+    this.props.wallet.getAddress().then(address => {
```

Comment:
Just to be clear, the wallet's `getAddress` function's behavior is to return the checksumm'd address, so there's no need to do it here. But that's good to have that rule laid out for where we do what with it.

As for the valid / invalid address, that's being handled in [`generateTransaction`](https://github.com/MyEtherWallet/MyEtherWallet/pull/100/files#diff-c762f03dadee1e4d3252d9aaa56e84a6R77)